### PR TITLE
add counter for deletion dropping optimization

### PR DESCRIPTION
**Pull request by ajkr**
add this counter stat to track usage of deletion-dropping optimization. if usage is low, we can delete it to prevent bugs like #2726.

Test Plan: unit test

### CODE REVIEW

**Comment by ajkr** [comment]
hm, if it's an internal stat, it won't be exposed anywhere. but if it's not internal, we can't delete it easily. what do others think we should do?

File: monitoring/statistics.cc [hunk 1]
```diff
@@ -18,7 +18,7 @@
 namespace rocksdb {
 
 std::shared_ptr<Statistics> CreateDBStatistics() {
-  return std::make_shared<StatisticsImpl>(nullptr, false);
+  return std::make_shared<StatisticsImpl>(nullptr, true);
```
Comment (sagar0) [inline monitoring/statistics.cc]: Does changing this existing default break any existing code?

File: db/compaction_iteration_stats.h [hunk 2]
```diff
@@ -16,6 +16,9 @@ struct CompactionIterationStats {
   int64_t num_record_drop_obsolete = 0;
   int64_t num_record_drop_range_del = 0;
   int64_t num_range_del_drop_obsolete = 0;
+  int64_t num_optimized_del_drop_obsolete = 0;  // Deletions obsoleted before
+                                                // bottom level due to file gap
+                                                // optimization.
```
Comment (sagar0) [inline db/compaction_iteration_stats.h]: Prefer to have the comment on a separate line above the field.

**Comment by ajkr** [comment]
I decided to make it externally visible because that's the only way we can collect it.

**Comment by sagar0** [comment]
Yup, I was wondering myself whether it can be collected if it is not visible externally.

File: db/compaction_iteration_stats.h (hunk 2 above)
Comment (ajkr) [inline db/compaction_iteration_stats.h]: will do

**Comment by andelf** [comment]
new TickersNameMap/Tikers enum items should be better added to the bottom instead of middle.

**Comment by ajkr** [comment]
good point.

**Pull request closed without merging**